### ISSUE TITLE
Pdf2Json hangs when loading a page with an image of the moon on it.

### ISSUE THREAD

**Issue by rsshilli** [modesty/pdf2json#182]
Hello,

Thank you for this library!  I actually use it through the PDFReader wrapper, but after trying to dig into why that library couldn't read my PDF, I realized this one was actually causing the issue.

I've developed the following minimal test:
```
const PDFParser = require("pdf2json/pdfparser");
const fs = require("fs");

new Promise((resolve, reject) => {
    const pdfParser = new PDFParser();
    pdfParser.on("pdfParser_dataError", (err) => {
        console.log("Error found reading data:", err);
        reject(err);
    });
    pdfParser.on("pdfParser_dataReady", function (pdfData) {
        console.log("I got some data!  It is: " + JSON.stringify(pdfData));
    });

    pdfParser.loadPDF("./SampleFile.pdf", 5);
});
```

I've attached the [SampleFile.pdf](https://github.com/modesty/pdf2json/files/2990349/SampleFile.pdf).  When I run this test, I see the following:
```
C:\projects\myproject> node doit.js
Info: about to load PDF file ./src/test/features/files/SampleUpload3.pdf
Info: Load PDF file status:Success!
Warning: Setting up fake worker.
Info: PDF loaded. pagesCount = 1
Info: start to parse page:1
Info: Reset: isSymbolicFont (false) for ArialMT
Info: NO_OP Begin: CanvasGraphics_beginAnnotations - 89
```

That last line is all that I see and then my little program waits forever.  It's hung, it seems.  Does anybody know of a fix or a workaround?

I'm using pdf2json@1.1.2. I created the PDF above by opening up a Google doc, typing that text and pasting in an image.  I was just creating a sample that I can use in test cases for my code.   I'm also on Windows 10 using Node 8.10.0, if that matters.

**Comment by spilliton** [comment]
We are also seeing this issue on one of our PDFs.  We are on on OSX pdf2json@1.1.8 Node v8.10.0

In addition when when Ctrl+C the process it appears to actually still be running and we we have `kill -9` it.  This part may not be related to pdf2json at all but figured I would mention it because it was strange.

**Comment by spilliton** [comment]
A couple more details we have discovered:

* We can repro with PDF documents that have "highlighted" text (see screenshot from Acrobat Pro DC).  If I delete the highlight the file parses correctly.
* When we parse the same file in our Amazon Lambda Node.js functions, it parses correctly (vs when I try locally on OSX).

<img width="794" alt="Screen Shot 2019-05-10 at 1 11 02 PM" src="https://user-images.githubusercontent.com/131613/57547894-30218c80-7325-11e9-83d5-ca02d941aa7f.png">

**Comment by blairjanis** [comment]
Has anyone had any luck with this issue.  I was seeing something similar and it turns out in my case the code is hanging on a page with highlights like @spilliton mentioned.  I spent hours digging into the issue and finally pinned down where the issue is.  In canvas.js there is a prototype for "paintFormXObjectEnd".  In this function the CanvasGraphics_paintFormXObjectEnd function there is a do ... while loop that apparently is handling an issue where some pdfs don't close all restores.  The "hanging" is happening in this this do ... while.  Despite the repetitive this.restore() call the depth is always 0 and this.current.paintFormXObjectDepth is always 0.  So the code is just sitting inside this eternal loop.  I have tried several things to try to get out of this loop and move on but if I remove the loop or try to skip it the program hangs somewhere else.  I've not figured out where yet.  Anyway, if anyone has any insight into how this can be avoided I would be very grateful.

```
    paintFormXObjectEnd: function CanvasGraphics_paintFormXObjectEnd() {
      var depth = this.current.paintFormXObjectDepth;
      do {
        this.restore();
        // some pdf don't close all restores inside object
        // closing those for them
      } while (this.current.paintFormXObjectDepth >= depth);
      this.baseTransform = this.baseTransformStack.pop();
    },
```

**Comment by blairjanis** [comment]
In case anyone runs across this problem I wanted to let you know I was able to find a solution.  Caveat, the solution works for me in the context of pdf2json because I don't care about images or highlighting.

In the pdf2json module there is a file in `/node_modules/pdf2json/base/shared/` called `util.js`.  On line 149 of this file (at least in the version I'm using (which I believe is the latest as of today) there is the following line of code:

```var NO_OPS_RANGE = PDFJS.NO_OPS_RANGE = [78, 79, 80, 81]; //range pairs, all ops with each pair will be skipped. !important!```

I modified this line to be ...

```var NO_OPS_RANGE = PDFJS.NO_OPS_RANGE = [74, 75, 78, 79, 80, 81]; //range pairs, all ops with each pair will be skipped. !important!```

I added the range pair of 74, 75.  This causes the renderer to skip `paintFormXObjectBegin` and `paintFormXObjectEnd` which I believe are the highlighting objects.  There may be other similar objects that will not be processed but again, in my use case I don't need these so it works for me.
